### PR TITLE
feat: added BaseRef-validator based on Title-validator

### PR DESCRIPTION
**Pull request by roydenwa**
BaseRef-validator for pull_requests to check "destination branch" with must_include and must_exclude conditions

### CODE REVIEW

**Comment by epDHowwD** [comment]
fixes: #343

File: lib/validators/baseRef.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      'pull_request_review.dismissed',
+      'pull_request.labeled',
+      'pull_request.milestoned',
+      'pull_request.demilestoned',
+      'pull_request.assigned',
+      'pull_request.unassigned',
+      'pull_request.unlabeled',
+      'pull_request.synchronize',
+      'pull_request.push_synchronize'
+    ]
```
Comment (jusx) [inline lib/validators/baseRef.js:20]: When all the actions of an event is supported we typically like to write this like so:
```js
this.supportedEvents = [
      'pull_request.*',
      'pull_request_review.*'
    ]
```

This way, it is understood that everything within pull_request and pull_request_review is supported. It is becomes consistent with the docs.

**Review by jusx — COMMENTED** [review]
LGTM -- will approve and merge after addressing the comment.

File: lib/validators/baseRef.js (hunk 1 above)
Comment (roydenwa) [reply lib/validators/baseRef.js:9]: @jusx makes sense, I changed it.

**Review by jusx — APPROVED** [review]

**Pull request merged**